packages/expo-sqlite/src/next/hooks.tsx

```diff
@@ -89,48 +170,118 @@ export function SQLiteProvider({
       databaseRef.current = null;
       setLoading(true);
     };
-  }, [dbName, options, initHandler]);
+  }, [databaseName, options, onInit]);
 
   if (error != null) {
     const handler =
-      errorHandler ??
+      onError ??
       ((e) => {
         throw e;
       });
     handler(error);
   }
-
   if (loading || !databaseRef.current) {
-    return loadingFallback != null ? <>{loadingFallback}</> : null;
+    return null;
   }
   return <SQLiteContext.Provider value={databaseRef.current}>{children}</SQLiteContext.Provider>;
 }
 
+function getDatabaseAsync({
+  databaseName,
+  options,
+  onInit,
+}: Pick<SQLiteProviderProps, 'databaseName' | 'options' | 'onInit'>): Promise<SQLiteDatabase> {
+  if (
+    databaseInstance?.promise != null &&
+    databaseInstance?.databaseName === databaseName &&
+    databaseInstance?.options === options &&
+    databaseInstance?.onInit === onInit
+  ) {
+    return databaseInstance.promise;
+  }
+
+  let promise: Promise<SQLiteDatabase>;
+  if (databaseInstance?.promise != null) {
+    promise = databaseInstance.promise
+      .then((db) => {
+        db.closeAsync();
+      })
+      .then(() => {
+        return openDatabaseWithInitAsync({ databaseName, options, onInit });
+      });
+  } else {
+    promise = openDatabaseWithInitAsync({ databaseName, options, onInit });
+  }
+  databaseInstance = {
+    databaseName,
+    options,
+    onInit,
+    promise,
+  };
+  return promise;
+}
+
+async function openDatabaseWithInitAsync({
+  databaseName,
+  options,
+  onInit,
+}: Pick<SQLiteProviderProps, 'databaseName' | 'options' | 'onInit'>): Promise<SQLiteDatabase> {
+  const database = await openDatabaseAsync(databaseName, options);
+  if (onInit != null) {
+    await onInit(database);
+  }
+  return database;
+}
+
+//#endregion
+
+//#region Private Suspense API similar to `React.use`
+
+// Referenced from https://github.com/vercel/swr/blob/1d8110900d1aee3747199bfb377b149b7ff6848e/_internal/src/types.ts#L27-L31
+type ReactUsePromise<T, E extends Error = Error> = Promise<T> & {
+  status?: 'pending' | 'fulfilled' | 'rejected';
+  value?: T;
+  reason?: E;
+};
+
+// Referenced from https://github.com/reactjs/react.dev/blob/6570e6cd79a16ac3b1a2902632eddab7e6abb9ad/src/content/reference/react/Suspense.md
 /**
- * A global hook for accessing the SQLite database across components.
- * This hook should only be used within a [`<SQLiteProvider>`](#sqliteprovider) component.
- *
- * @example
- * ```tsx
- * export default function App() {
- *   return (
- *     <SQLiteProvider dbName="test.db">
- *       <Main />
- *     </SQLiteProvider>
- *   );
- * }
- *
- * export function Main() {
- *   const db = useSQLiteContext();
- *   console.log('sqlite version', db.getSync('SELECT sqlite_version()'));
- *   return <View />
- * }
- * ```
+ * A custom hook like `React.use` hook using private Suspense implementation.
```

Comment:
```suggestion
 * A custom hook like [`React.use`](https://react.dev/reference/react/use) hook using private Suspense implementation.
```

Let's add a link to the React definition here.